WireDomain/Sources/WireDomain/Event Processing/UserEventProcessor/UserLegalholdEnableEventProcessor.swift

```diff
@@ -32,9 +33,51 @@ protocol UserLegalholdEnableEventProcessorProtocol {
 
 struct UserLegalholdEnableEventProcessor: UserLegalholdEnableEventProcessorProtocol {
 
-    func processEvent(_: UserLegalholdEnableEvent) async throws {
-        // TODO: [WPB-10195]
-        assertionFailure("not implemented yet")
+    let context: NSManagedObjectContext
+    let userRepository: any UserRepositoryProtocol
+    let clientRepository: any ClientRepositoryProtocol
+
+    func processEvent(_ event: UserLegalholdEnableEvent) async throws {
+        let userID = event.userID
+
+        let selfUserID = await context.perform {
+            let selfUser = userRepository.fetchSelfUser()
+            return selfUser.remoteIdentifier
+        }
+
+        guard userID == selfUserID else {
+            return
+        }
+
+        try await processSelfUserClients()
     }
 
+    /// Fetches, creates and updates clients for self user and removes the deleted clients locally.
+
+    private func processSelfUserClients() async throws {
+        let remoteSelfClients = try await clientRepository.fetchSelfClients()
+        let localSelfClients = await context.perform {
+            let selfUser = userRepository.fetchSelfUser()
+            return selfUser.clients
+        }
+
+        for remoteSelfClient in remoteSelfClients {
+            let localUserClient = try await clientRepository.fetchOrCreateClient(
+                with: remoteSelfClient.id
+            )
+
+            try await clientRepository.updateClient(
+                with: remoteSelfClient.id,
+                from: remoteSelfClient,
+                isNewClient: localUserClient.isNew
+            )
+        }
+
+        let deletedSelfClientsIDs = localSelfClients.compactMap(\.remoteIdentifier).filter { !remoteSelfClients.map(\.id).contains($0)
+        }
+
+        for deletedSelfClientID in deletedSelfClientsIDs {
+            await clientRepository.deleteClient(with: deletedSelfClientID)
+        }
+    }
```

Comment:
I think this should all be inside the client repository.

Reply:
done in https://github.com/wireapp/wire-ios/pull/2019/commits/d0e4a57aa3390d8146ceda7f9ce8dd4ac87cf0b4 , I moved this to `pullSelfClients` . Maybe we should rename our pull methods to sync methods like `syncSelfClients` since we do more than just pulling here: fetching from API, storing/updating clients clients locally and deleting no longer relevant clients, basically syncing our local store with backend. What do you think ?